### ISSUE TITLE
Peer group manager where communicator fails to start is recoverable

### ISSUE THREAD

**Issue by lbradstreet** [onyx-platform/onyx#862]
Following error causes Onyx to exit, as the peer group failed to start up again, however this error is recoverable.

```
18-05-14 01:35:32 n2 ERROR [onyx.peer.peer-group-manager:366] - Unrecoverable error caught in PeerGroupManager loop. Exiting.
                                                         java.lang.Thread.run                            Thread.java:  745
                           java.util.concurrent.ThreadPoolExecutor$Worker.run                ThreadPoolExecutor.java:  617
                            java.util.concurrent.ThreadPoolExecutor.runWorker                ThreadPoolExecutor.java: 1142
                                                                          ...                                             
                                            clojure.core.async/thread-call/fn                              async.clj:  441
                             onyx.peer.peer-group-manager.PeerGroupManager/fn                 peer_group_manager.clj:  400
                         onyx.peer.peer-group-manager/peer-group-manager-loop                 peer_group_manager.clj:  363
                                                                          ...                                             
                                    onyx.peer.peer-group-manager/eval12587/fn                 peer_group_manager.clj:  301
                                                                          ...                                             
                                    onyx.peer.peer-group-manager/eval12477/fn                 peer_group_manager.clj:  165
                                                                          ...                                             
                                    onyx.peer.peer-group-manager/eval12442/fn                 peer_group_manager.clj:   97
                                                                          ...                                             
                                    onyx.peer.peer-group-manager/eval12433/fn                 peer_group_manager.clj:   78
                             onyx.peer.peer-group-manager/start-communicator!                 peer_group_manager.clj:   69
                                        onyx.peer.communicator.OnyxComm/start                       communicator.clj:   85
                                      com.stuartsierra.component/start-system                         component.cljc:  163
                                                                          ...                                             
                                     com.stuartsierra.component/update-system                         component.cljc:  129
                                     com.stuartsierra.component/update-system                         component.cljc:  135
                                                          clojure.core/reduce                               core.clj: 6544
                                                                          ...                                             
                                  com.stuartsierra.component/update-system/fn                         component.cljc:  139
                                        com.stuartsierra.component/try-action                         component.cljc:  117
                                                           clojure.core/apply                               core.clj:  648
                                                                          ...                                             
                                              com.stuartsierra.component/fn/G                         component.cljc:    5 (repeats 2 times)
                                           onyx.log.zookeeper.ZooKeeper/start                          zookeeper.clj:  130
                                                                          ...                                             
                                                      onyx.log.curator/create                            curator.clj:   85
                                                      onyx.log.curator/create                            curator.clj:   93
                  org.apache.curator.framework.imps.CreateBuilderImpl.forPath                 CreateBuilderImpl.java:   49
                  org.apache.curator.framework.imps.CreateBuilderImpl.forPath                 CreateBuilderImpl.java:  573
                  org.apache.curator.framework.imps.CreateBuilderImpl.forPath                 CreateBuilderImpl.java:  595
org.apache.curator.framework.imps.CreateBuilderImpl.protectedPathInForeground                 CreateBuilderImpl.java:  605
         org.apache.curator.framework.imps.CreateBuilderImpl.pathInForeground                 CreateBuilderImpl.java: 1155
                                   org.apache.curator.RetryLoop.callWithRetry                         RetryLoop.java:  100
 org.apache.curator.connection.StandardConnectionHandlingPolicy.callWithRetry  StandardConnectionHandlingPolicy.java:   64
                  org.apache.curator.framework.imps.CreateBuilderImpl$17.call                 CreateBuilderImpl.java: 1158
                  org.apache.curator.framework.imps.CreateBuilderImpl$17.call                 CreateBuilderImpl.java: 1177
                                        org.apache.zookeeper.ZooKeeper.create                         ZooKeeper.java:  786
                                  org.apache.zookeeper.KeeperException.create                   KeeperException.java:   54
                                  org.apache.zookeeper.KeeperException.create                   KeeperException.java:  102
org.apache.zookeeper.KeeperException$ConnectionLossException: KeeperErrorCode = ConnectionLoss for /onyx
    code: -4
    path: "/onyx"
                                  clojure.lang.ExceptionInfo: Error in component :log in system onyx.peer.communicator.OnyxComm calling #'com.stuartsierra.component/start
     component: #<ZooKeeper Component>
      function: #'com.stuartsierra.component/start
        reason: :com.stuartsierra.component/component-function-threw-exception
        system: #onyx.peer.communicator.OnyxComm{:config {:onyx.bookkeeper/ledger-password "INSECUREDEFAULTPASSWORD", :onyx/tenancy-id "JEPSENONYXID", :onyx.bookkeeper/client-throttle 30000, :onyx.peer/storage :s3, :onyx.zookeeper/backoff-max-retries 4, :onyx.bookkeeper/client-timeout 60000, :onyx.bookkeeper/zk-ledgers-root-path "/ledgers", :onyx.query.server/ip "172.18.0.3", :onyx.peer/storage.s3.region "us-west-2", :onyx.messaging/impl :aeron, ...}, :logging-config #<Logging Configuration>, :monitoring #onyx.monitoring.metrics_monitoring.Monitoring{:monitoring :custom, :zookeeper-read-origin #object[onyx.monitoring.metrics_monitoring.Monitoring$fn__6805 0x5f482f15 "onyx.monitoring.metrics_monitoring.Monitoring$fn__6805@5f482f15"], :zookeeper-write-messaging #object[onyx.monitoring.metrics_monitoring.Monitoring$fn__6785 0x11ccbbde "onyx.monitoring.metrics_monitoring.Monitoring$fn__6785@11ccbbde"], :group-notify-join #object[onyx.monitoring.metrics_monitoring.Monitoring$fn__6815 0x587afe33 "onyx.monitoring.metrics_monitoring.Monitoring$fn__6815@587afe33"], :zookeeper-write-catalog #object[onyx.monitoring.metrics_monitoring.Monitoring$fn__6771 0x6f7cb52d "onyx.monitoring.metrics_monitoring.Monitoring$fn__6771@6f7cb52d"], :zookeeper-write-workflow #object[onyx.monitoring.metrics_monitoring.Monitoring$fn__6773 0x4c3029d "onyx.monitoring.metrics_monitoring.Monitoring$fn__6773@4c3029d"], :zookeeper-force-write-chunk #object[onyx.monitoring.metrics_monitoring.Monitoring$fn__6787 0x42b4c004 "onyx.monitoring.metrics_monitoring.Monitoring$fn__6787@42b4c004"], :peer-error! #object[onyx.monitoring.metrics_monitoring.Monitoring$fn__6767 0x6a79273d "onyx.monitoring.metrics_monitoring.Monitoring$fn__6767@6a79273d"], :zookeeper-read-log-entry #object[onyx.monitoring.metrics_monitoring.Monitoring$fn__6791 0x18db9cf7 "onyx.monitoring.metrics_monitoring.Monitoring$fn__6791@18db9cf7"], :registry #object[com.codahale.metrics.MetricRegistry 0x7f5d965f "com.codahale.metrics.MetricRegistry@7f5d965f"], ...}, :log #<ZooKeeper Component>, :replica-subscription #onyx.peer.communicator.ReplicaSubscription{:peer-config {:onyx.bookkeeper/ledger-password "INSECUREDEFAULTPASSWORD", :onyx/tenancy-id "JEPSENONYXID", :onyx.bookkeeper/client-throttle 30000, :onyx.peer/storage :s3, :onyx.zookeeper/backoff-max-retries 4, :onyx.bookkeeper/client-timeout 60000, :onyx.bookkeeper/zk-ledgers-root-path "/ledgers", :onyx.query.server/ip "172.18.0.3", :onyx.peer/storage.s3.region "us-west-2", :onyx.messaging/impl :aeron, ...}}, :log-writer #onyx.peer.communicator.LogWriter{:peer-config {:onyx.bookkeeper/ledger-password "INSECUREDEFAULTPASSWORD", :onyx/tenancy-id "JEPSENONYXID", :onyx.bookkeeper/client-throttle 30000, :onyx.peer/storage :s3, :onyx.zookeeper/backoff-max-retries 4, :onyx.bookkeeper/client-timeout 60000, :onyx.bookkeeper/zk-ledgers-root-path "/ledgers", :onyx.query.server/ip "172.18.0.3", :onyx.peer/storage.s3.region "us-west-2", :onyx.messaging/impl :aeron, ...}, :group-ch #object[clojure.core.async.impl.channels.ManyToManyChannel 0x3dcb54 "clojure.core.async.impl.channels.ManyToManyChannel@3dcb54"]}}
    system-key: :log
```

**Comment by lbradstreet** [comment]
Fixed in 0.13.0.

**Comment by gokulreddy07** [comment]
I'm facing same issue in onyx 0.13.0. version.
